mlos_bench/mlos_bench/launcher.py

```diff
@@ -146,6 +146,8 @@ def __init__(self, description: str, long_text: str = "", argv: Optional[List[st
         _LOG.info("Init storage: %s", self.storage)
 
         self.teardown: bool = bool(args.teardown) if args.teardown is not None else bool(config.get("teardown", True))
```

Comment:
Should `self.teardown` also become `global_config['teardown']`?

I'm a little wary about moving some of these things into `global_config`.

`trial_id` makes sense, but does `trial_config_repeat_config`?
When would a user script need to access that value (e.g., via `shell_env_params`)?

Also, we should document somewhere the set of "known" variables that go into `global_config` so that 
a) users don't conflict with them
b) they know what they can use in their scripts

Reply:
technically, it should, but if you add `teardown` to global config, it will override the `teardown` property of all `ScriptEnv` environments :)

Reply:
but in all, yes, that's more of a backward compatibility fix so we can keep using e.g. '--trial-config-repeat-count` option instead of `--trial_config_repeat_count` (with underscores)